File: conformance/bundles/10-error-fence-without-step/CukesSteps.java
package dev.varar.conformance.bundle10;

import dev.varar.Registrar;
import dev.varar.State;
import dev.varar.StateBinder;
import dev.varar.StepDefinitions;

/**
 * Java sibling of {@code cukes.steps.ts} / {@code cukes.steps.py} (bundle {@code
 * 10-error-fence-without-step}). The example's prose matches no step, so the {@code
 * error} fence (which marks the example expected-to-fail) has nothing to run ->
 * error-fence-without-step diagnostic, and the example is dropped (a plan-stage
 * concern; this stage only needs the one step registered).
 */
public final class CukesSteps implements StepDefinitions {

    record Ctx() implements State {}

    @Override
    public void defineSteps(Registrar registrar) {
        StateBinder<Ctx> s = registrar.steps(Ctx::new);

        s.stimulus("I have {int} cukes", (Ctx ctx, Integer n) -> ctx);
    }
}

File: conformance/bundles/10-error-fence-without-step/cukes.steps.kt
@file:JvmName("CukesSteps")

// Kotlin sibling of cukes.steps.ts / cukes.steps.py / CukesSteps.java (bundle
// 10-error-fence-without-step): the example's prose matches no step, so the
// error fence has nothing to run — a plan-stage diagnostic; this stage only
// needs the one step registered.
package dev.varar.kotlin.conformance.bundle10

import dev.varar.kotlin.stimulus
import dev.varar.kotlin.steps

class Ctx

val steps = steps(::Ctx) {
    stimulus("I have {int} cukes") { n: Int -> this }
}
